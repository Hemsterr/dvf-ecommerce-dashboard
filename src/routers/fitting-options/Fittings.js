// @flow
import React from 'react'
import BookingItem from '../../components/BookingItem'
import { fittingOptions } from '../../constants/fittingOptions'

type Props = {
  handleSelectOption: Function,
}

class BookingItems extends React.PureComponent<Props> {
  static defaultProps = {
    handleSelectOption: () => {},
  }

  render() {
    const { handleSelectOption } = this.props
    return (
      <div className="fitting-options__wrapper">
        {fittingOptions.map(item => (
          <BookingItem
            key={item.type}
            type={item.type}
            description={item.description}
            handleSelectOption={handleSelectOption}
          />
        ))}
      </div>
    )
  }
}

export default BookingItems
