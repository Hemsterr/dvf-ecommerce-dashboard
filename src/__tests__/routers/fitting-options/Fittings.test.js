// Libs
import React from 'react'
import { render } from '@testing-library/react'

// Component
import Fittings from '../../../routers/fitting-options/Fittings'

const wrapper = (props = {}) => render(<Fittings {...props} />)

describe('Fittings component', () => {
  test('Render the Fittings component with default props', () => {
    const { container } = wrapper()
    expect(container).toMatchSnapshot()
    expect(Fittings.defaultProps.handleSelectOption()).toBeUndefined()
    expect(container.querySelectorAll('.bookingItem').length).toEqual(3)
  })
})
