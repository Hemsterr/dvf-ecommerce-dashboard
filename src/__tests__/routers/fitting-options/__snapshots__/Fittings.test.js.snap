// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fittings component Render the Fittings component with default props 1`] = `
<div>
  <div
    class="fitting-options__wrapper"
  >
    <div
      class="bookingItem bookingItem__fitting-kit"
      role="presentation"
    >
      <div>
        <span
          class="bookingItem__icon"
        />
      </div>
      <p
        class="bookingItem__title"
      >
        I would like to fit myself using Hemster's Fitting Kit
      </p>
    </div>
    <div
      class="bookingItem bookingItem__fit-stylist"
      role="presentation"
    >
      <div>
        <span
          class="bookingItem__icon"
        />
      </div>
      <p
        class="bookingItem__title"
      >
        I would like to book an appointment to meet with a Fit Stylist virtually
      </p>
    </div>
    <div
      class="bookingItem bookingItem__in-home-stylist"
      role="presentation"
    >
      <div>
        <span
          class="bookingItem__icon"
        />
      </div>
      <p
        class="bookingItem__title"
      >
        I would like to book an In-Home Appointment with a Fit Stylist
      </p>
    </div>
  </div>
</div>
`;
